browser/components/MarkdownPreview.js

```diff
@@ -338,6 +348,14 @@ export default class MarkdownPreview extends React.Component {
     e.stopPropagation()
   }
 
+  handlelinkClick (e) {
+    const noteHash = e.target.hash
+    const regexIsNoteLink = /^#(.{20})-(.{20})$/
```

Comment:
Can you cut out this logic into the module?
If you do it, you can write unit test for the logic. (And I think this is business logic.)